### PR TITLE
fix(select): updates onChange prop to be optional

### PR DESCRIPTION
**Pull request by akornmeier**
- **update**: `onChange` prop to be optional
- **update**: removes default value from function signature to be assigned later if prop value is nullish

Closes issue: #389

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -3129,6 +3129,13 @@ create-hmac@^1.1.0, create-hmac@^1.1.2, create-hmac@^1.1.4:
     safe-buffer "^5.0.1"
     sha.js "^2.4.8"
 
+cross-env@^7.0.3:
```
Comment (schne324) [inline yarn.lock:3132]: all of this diff seems weird but I remember you told me yesterday that there were agora deps checked into source? I'm not seeing those removed here...I'm also not seeing any agora deps in the [`yarn.lock` in the develop branch](https://raw.githubusercontent.com/dequelabs/cauldron/develop/yarn.lock) either 🤔

Comment (akornmeier) [reply yarn.lock:3132]: They were from my initial commit. I removed them by uninstalling and reinstalling the cross-env package

Comment (akornmeier) [reply yarn.lock:3132]: i had my Deque npmrc pointed to agora when I checked in the first time.

Comment (schne324) [reply yarn.lock:3132]: so this patch doesn't update any deps but has yarn.lock diffs?

**Review by schne324 — APPROVED** [review]
Not gonna hold this up for some yarn.lock nonsense...ideally we back those changes out though 🤷

File: yarn.lock (hunk 1 above)
Comment (akornmeier) [reply yarn.lock]: @schne324 - I reverted the `yarn.lock` changes that were caused by my DQ .npmrc config. I will need a re-thumb.

**Review by schne324 — APPROVED** [review]

**Pull request merged**